### PR TITLE
adds support for re-exporting crate

### PR DESCRIPTION
**Pull request by colstrom**
Heyo! Thanks for making `veil`, it's useful!

I'd like to request (and provide) a small change that makes it possible to use `veil` in a way that it doesn't currently support.

As it stands, the `::veil` paths are anchored, which requires the depending crate to directly depend on `veil`. That's suitable for most cases (clearly, since it only works this way currently), but it also makes it so that importing it from a crate that re-exports it doesn't work. This in turn makes it difficult to provide a batteries-included sort of functionality in downstream crates.

This change removes the path anchors, and to the best of my knowledge should not break anything, except what I assume would be an unusual scenario where another module exists in the same scope that also happens to be named `veil`. This puts the responsibility on whoever does _that_ to anchor the reference **there**, which seems like a fair trade.

The test suite still runs (and passes!), and I haven't done anything with the versioning, since while I'm pretty sure that this won't cause any issues, I'm still not quite clear on how that would intersect with versioning. 🤔

If you'd like me to change anything, just let me know and I'd be happy to do my best!

Thanks again, people like you making cool things is why we have so much great stuff in the Rust ecosystem. ❤️

### CODE REVIEW

**Review by MaeIsBad — DISMISSED** [review]
Seems fine, thanks :slightly_smiling_face: 

I'll wait for another dev to review before merging

File: CHANGELOG.md
Comment (cottinisimone) [inline CHANGELOG.md:14]: veil imports?

**Review by cottinisimone — APPROVED** [review]
It was cleaner to open a new PR for fixing clippy things but.. 
The rest LGTM

**Comment by MaeIsBad** [comment]
> It was cleaner to open a new PR for fixing clippy things but..

yeah probably

**Review by MaeIsBad — APPROVED** [review]

**Review by MaeIsBad — APPROVED** [review]

**Review by MaeIsBad — APPROVED** [review]

**Pull request merged**
